language/bytecode-verifier/src/signature.rs

```diff
@@ -251,23 +300,28 @@ impl<'a> SignatureChecker<'a> {
                 // i.e. it cannot be checked unless we are inside some module member. The only case
                 // where that happens is when checking the signature pool itself
                 let sh = self.resolver.struct_handle_at(*idx);
-                self.check_generic_instance(type_arguments, &sh.type_parameters, type_parameters)
+                self.check_generic_instance(
+                    type_arguments,
+                    sh.type_param_constraints(),
+                    type_parameters,
+                )
             }
             _ => Ok(()),
         }
     }
 
     // Checks if the given types are well defined and satisfy the constraints in the given context.
-    fn check_generic_instance(
+    fn check_generic_instance<'b>(
         &self,
         type_arguments: &[SignatureToken],
-        constraints: &[AbilitySet],
+        constraints: impl IntoIterator<Item = &'b AbilitySet>,
```

Comment:
Nit, just copy the ability set. It will always be faster since it is just a u8 (I think)